sdk/core/AzureCore/Source/DataStructures/AzureDate.swift

```diff
@@ -0,0 +1,150 @@
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Foundation
+
+public struct Iso8601Date: RequestStringConvertible, Codable, Equatable, Comparable {
+    static var dateFormat: AzureDateFormat = .iso8601
+
+    static var formatter: DateFormatter {
+        return Self.dateFormat.formatter
+    }
+
+    public var date: Date
+
+    // MARK: RequestStringConvertible
+
+    public var requestString: String {
+        return Self.formatter.string(from: date)
+    }
+
+    // MARK: Initializers
+
+    public init() {
+        self.date = Date()
+    }
+
+    public init?(string: String?) {
+        guard let date = Self.formatter.date(from: string ?? "") else { return nil }
+        self.date = date
+    }
+
+    public init?(_ date: Date?) {
+        guard let unwrapped = date else { return nil }
+        self.date = unwrapped
+    }
+
+    // MARK: Equatable
+
+    public static func == (lhs: Iso8601Date, rhs: Iso8601Date) -> Bool {
+        return lhs.requestString == rhs.requestString
+    }
+
+    // MARK: Comparable
+
+    public static func < (lhs: Iso8601Date, rhs: Iso8601Date) -> Bool {
+        return lhs.date < rhs.date
+    }
+}
+
+public struct Rfc1123Date: RequestStringConvertible, Codable, Equatable, Comparable {
+    static var dateFormat: AzureDateFormat = .rfc1123
+
+    static var formatter: DateFormatter {
+        return Self.dateFormat.formatter
+    }
+
+    public var date: Date
+
+    // MARK: RequestStringConvertible
+
+    public var requestString: String {
+        return Self.formatter.string(from: date)
+    }
+
+    // MARK: Initializers
+
+    public init() {
+        self.date = Date()
+    }
+
+    public init?(string: String?) {
+        guard let date = Self.formatter.date(from: string ?? "") else { return nil }
+        self.date = date
+    }
+
+    public init?(_ date: Date?) {
+        guard let unwrapped = date else { return nil }
+        self.date = unwrapped
+    }
+
+    // MARK: Equatable
+
+    public static func == (lhs: Rfc1123Date, rhs: Rfc1123Date) -> Bool {
+        return lhs.requestString == rhs.requestString
+    }
+
+    // MARK: Comparable
+
+    public static func < (lhs: Rfc1123Date, rhs: Rfc1123Date) -> Bool {
+        return lhs.date < rhs.date
+    }
+}
+
+public enum AzureDateFormat {
+    /// Format a custom date format
+    case custom(String)
+
+    /// Format the `Date` as an RFC 1123-formatted string.
+    case rfc1123
+
+    /// Format the `Date` as an ISO-8601-formatted string (in RFC 3339 format).
+    case iso8601
+
+    /// Retrieve a `DateFormatter` for the given date format.
+    public var formatter: DateFormatter {
+        switch self {
+        case let .custom(format):
+            let formatter = DateFormatter()
+            formatter.dateFormat = format
+            return formatter
+        case .rfc1123:
```

Comment:
Azure core already has these date formatter defined. Can we resue them? https://github.com/Azure/azure-sdk-for-ios/blob/65a2dba0447ba20d58e74af0f96ce2a2f123c8d3/sdk/core/AzureCore/Source/Util/ConvertingInitializers.swift

Reply:
They have been relocated here.